asteroid/src/main.py

```diff
@@ -175,12 +193,29 @@ def process_game_logic(self):
         self.ship.move()
         self.ship.slow_down(CONSTANT_DECELERATION)
         self.ship.keep_within_borders()
+        self.move_projectiles()
 
     def draw_game_elements(self):
         self.screen.fill((0, 0, 0))
         self.screen.blit(self.ship.displayed_sprite, self.ship.pos)
+        self.draw_projectiles()
         pygame.display.flip()
 
+    def move_projectiles(self):
+        for p in self.projectiles:
+                    if p.pos.x  < 0 or p.pos.x > SCREEN_WIDTH:
+                        p.cleanup = True
+                    elif p.pos.y < 0 or p.pos.y > SCREEN_HEIGHT:
+                        p.cleanup = True
+                    p.pos += p.velocity_vector
+
+    def cleanup_projectiles(self):

```

Comment:
This could be consolidated into the move and/or draw step as well

Reply:
use your best judgement with the the most important factors being readability and simplicity.